include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -153,7 +153,17 @@ class kmer_hash_view : public std::ranges::view_interface<kmer_hash_view<urng_t>
      */
     auto end() noexcept
     {
-        return std::ranges::end(urange);
+        if constexpr (std::ranges::common_range<urng_t>)
```

Comment:
Hm.. Why don't we always return a shape iterator then? Does the urange sentinel have any advantages?

Reply:
Yes, apparently we can just guarantee common_range; I did something before where it did not work, but now it works.

Reply:
No, we can't, because in order to retain the semantics of an end_iterator, we need to be able to assign the end of the underlying range to the `text_right` iterator of the `shape_iterator`. This only works if both have a common type, i.e. the range is a common range.

Reply:
Thanks for explaining!